### PR TITLE
Update MarkBind documentation to use layouts

### LINKED ISSUE #527: Apply a default layout for the MarkBind website
Suggestion: use a default layout for the MarkBind website

### PR DESCRIPTION
**Pull request by Chng-Zhi-Xuan**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Documentation update

Resolves #527 

**What is the rationale for this request?**

We can clean up the `<frontmatter>` of each page, within our documentation by using the layouts feature. Since #528 was fixed, we can proceed to use layouts with no issues.

**What changes did you make? (Give an overview)**

Replaced the `header`, `footer` and `siteNav` values of `<frontmatter>` with `layout`

**Testing instructions:**

- Open Netlify preview and check that the documentation is rendered fine.

**Proposed commit message: (wrap lines at 72 characters)**

**Docs: update pages' frontmatter to use layouts**

MarkBind documentation pages' frontmatter currently specifies 3 values
to insert headers, footers and site navigation. The issue with layouts
inserting siteNav CSS files has been fixed in #841 .

Therefore, let's update our documentation to use the layouts feature to
reduce the clutter in each pages' frontmatter. A separate layout for 
both user guide and developer guide has been added.

### CODE REVIEW

File: docs/_markbind/layouts/devGuide/header.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<header>
+  <link rel="stylesheet" href="{{baseUrl}}/css/main.css">
+  <navbar type="dark">
+    <a slot="brand" href="{{baseUrl}}/index.html" title="Home" class="navbar-brand"><img src="{{baseUrl}}/images/logo-darkbackground.png" height="20" /></a>
```
Comment (openorclose) [inline docs/_markbind/layouts/devGuide/header.md]: Minor nit: remove the backlash in img to be in line with the other img tags

File: docs/_markbind/layouts/userGuide/header.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<header>
+  <link rel="stylesheet" href="{{baseUrl}}/css/main.css">
+  <navbar type="dark">
+    <a slot="brand" href="{{baseUrl}}/index.html" title="Home" class="navbar-brand"><img src="{{baseUrl}}/images/logo-darkbackground.png" height="20" /></a>
```
Comment (openorclose) [inline docs/_markbind/layouts/userGuide/header.md]: Minor nit: remove the backlash in img to be in line with the other img tags

File: docs/_markbind/layouts/devGuide/header.md (hunk 1 above)
Comment (openorclose) [reply docs/_markbind/layouts/devGuide/header.md]: https://www.w3.org/TR/html5/syntax.html#writing-html-documents-elements

The / character is redundant, and the rest of the code doesn't use / in img tags as well as well.

**Comment by Chng-Zhi-Xuan** [comment]
>Will header.md, footer.md, userGuideSections.md and devGuideSections.md be used after this PR is merged? If not, can consider removing it...

Some of the main pages (`docs/index.md`) do not have a layout applied to them as they do not specify a site navigation menu. Thus need to specify `header.md` and `footer.md` specifically.

As for `userGuideSections.md` and `devGuideSections.md`, note that within the layouts folder, both are named as `navigation.md` due to layouts feature implementation. I wanted to keep the original files where they were before in case we have to author different layouts in the future. Not having them there would require us to skim through all the different `navigation.md` within the layouts folder and picking the right one.

**Pull request merged**